azurefunctions-extensions-bindings-blob/azurefunctions/extensions/bindings/blob/utils.py

```diff
@@ -39,12 +50,52 @@ def get_connection_string(connection_string: str) -> str:
         )
 
 
-def using_managed_identity(connection_name: str) -> bool:
+def using_system_managed_identity(connection_name: str) -> bool:
     """
-    To determine if managed identity is being used, we check if the provided
-    connection string has either of the two suffixes:
+    To determine if system-assigned managed identity is being used, we check if
+    the provided connection string has either of the two suffixes:
     __serviceUri or __blobServiceUri.
     """
     return (os.getenv(connection_name + "__serviceUri") is not None) or (
         os.getenv(connection_name + "__blobServiceUri") is not None
     )
+
+
+def using_user_managed_identity(connection_name: str) -> bool:
+    """
+    To determine if user-assigned managed identity is being used, we check if
+    the provided connection string has the following suffixes:
+    __credential AND __clientId
+    """
+    return (os.getenv(connection_name + "__credential") is not None) and (
+        os.getenv(connection_name + "__clientId") is not None
+    )
+
+
+def get_blob_service_client(system_managed_identity: bool,
+                            user_managed_identity: bool,
```

Comment:
NIT: Better name would  be ServiceClientFactory.

Also, all these checks    using_system_managed_identity and using_user_managed_identity, can be moved in this method.